k8up/Makefile

```diff
@@ -0,0 +1,48 @@
+MAKEFLAGS += --warn-undefined-variables
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c
+.DEFAULT_GOAL := help
+.DELETE_ON_ERROR:
+.SUFFIXES:
+
+# Git-treeish of the K8up archive version to fetch
+K8UP_VERSION := master
+CRD_FILLE    := templates/crds.yaml
+
+SED_CMD := gsed
+
+# TODO Use the aggregated CRDs that will be attached to a release.
+# This will get rid of creating and removing the temporary `crds` folder.
+#
+# This is made phony on purpose. We do not have a local file that would trigger a rebuild.
+.PHONY: templates/crds.yaml
+templates/crds.yaml: ## Update the CRDs by fetching them from the k8up repository.
+	mkdir -p crds
+	curl -sL https://github.com/vshn/k8up/archive/$(K8UP_VERSION).tar.gz | tar -xj --directory crds --strip-components=4 k8up-master/config/crd/apiextensions.k8s.io/
+	echo -n "{{ if Capabilities.KubeVersion.Major == 1 && Capabilities.KubeVersion.Minor < 16 }}" > $@
+	# curl -sL https://github.com/vshn/k8up/releases/download/$(K8UP_VERSION)/k8up-crd-v1beta1.yaml >> $@
+	cat crds/v1beta1/* >> $@
+	echo -n "{{ else }}" >> $@
```

Comment:
Instead of a really big file with if-else clause and multiple CRD versions in it, I suggest to create separate files per CRD version still. If each file has its own if-clause in it, the effect of if-else is the same, but it makes the chart easier to understand.